controllers/registeredservice_controller.go

```diff
@@ -82,7 +120,7 @@ func (r *RegisteredServiceReconciler) Reconcile(ctx context.Context, req ctrl.Re
 	rs.Status = status
 
 	log.Info("Updating status of RegisteredService")
-	err := r.Status().Update(context.Background(), &rs)
+	err = r.Status().Update(context.Background(), &rs)
```

Comment:
Don't make a new context here; use the context you get passed from controller-runtime.
```suggestion
	err = r.Status().Update(ctx, &rs)
```

Reply:
OK